src/lib/ops.py

```diff
@@ -27,14 +29,14 @@ def __str__(self):
     s += super().__str__()
     return s
 
-  def dump(self, description=None, zeros=False):
+  def dump(self, description: Optional[str]=None, zeros: bool=False) -> None:
     res = ''
     if description:
       res += f'{description} ({self.nbits}-qubits operator)\n'
     for row in range(self.shape[0]):
       for col in range(self.shape[1]):
         val = self[row, col]
-        res += '{:+.1f}{:+.1f}i  '.format(val.real, val.imag)
+        res += f'{val.real:+.1f}{val.imag:+.1f}i  '
```

Comment:
Woussa. I have to admit, I was not aware of this syntax. Very neat! Thanks for the typing as well.